spec/ndx-patterned-ogen.extensions.yaml

```
@@ -122,45 +169,80 @@ groups:
         dtype: text
         doc: The model specification of the spatial light modulator (e.g. 'NeuraLight 3D Ultra', from Bruker).
         required: false
+    datasets:
       - name: spatial_resolution
         dtype: numeric
         doc: Resolution of spatial light modulator (in pixels), formatted as [width, height, depth].
-        required: false
+        quantity: "?"
+        dims:
+        - width, height, depth
+        shape:
+        - 3
   - neurodata_type_def: LightSource
     neurodata_type_inc: Device
     doc: Light source used in the experiment.
-    attributes:
+    datasets:
       - name: stimulation_wavelength
         dtype: numeric
         doc: Excitation wavelength of stimulation light (nanometers).
-        required: true
-      - name: model
-        dtype: text
-        doc: Model of light source device.
-        required: false
+        attributes:
+        - name: unit
+          value: nanometers
+          doc: unit of measure of wavelength which is fixed to nanometers
```

```suggestion
          doc: Unit of measure of wavelength, fixed to nanometers.
```

Also adjusted to be consistent with above